torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -281,15 +286,33 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                     f"FSDP only supports 1D TP, not {self._tp_spec.placements}"
                 )
             # TODO: Hard code FSDP + TP; need to support HSDP + TP
+            split_factor = self._tp_spec.num_shards_map[0]
             self._spmd_placements: Tuple[Placement, ...] = (
-                Shard(0),
+                (
+                    _StridedShard(0, split_factor=split_factor)
+                    if split_factor > 1
+                    else Shard(0)
+                ),
                 self._tp_spec.placements[0],
             )
             self._sharding_spec = DTensorSpec(
                 self._spmd_mesh,
                 self._spmd_placements,
                 tensor_meta=self._tp_spec.tensor_meta,
             )
+            # NOTE: FSDP+TP does not support uneven sharding for now
```

Comment:
hmmm I think this check is too strict, we should ONLY check the tensor dimension 0 for uneven sharding when the TP sharding spec sharded on tensor dim 0, as:
1. DTensor supports uneven sharding computation for its normal case.
2. Only when strided sharding enabled we don't yet support uneven sharding, and the strided sharding only enabled when the TP's DTensor is sharded on tensor dim 0 (and FSDP2 shards on tensor dim 0 too, which is the case where StridedShard comes in)

So please fix this check to be more accurate

Reply:
thanks for catching the mistake!! I added a check on `split_factor` and only perform the sharding evenness check when it's larger than `1` (i.e. tensor dim 0 has been sharded on TP mesh dim).